### PR TITLE
Autoowners: Add support for custom owners{,aliases} filenames

### PR DESCRIPTION
**Pull request by alvaroaleman**
/cc @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

File: cmd/autoowners/main_test.go [hunk 1]
```diff
@@ -5,15 +5,19 @@ import (
...
 	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins/ownersconfig"
 	"k8s.io/test-infra/prow/repoowners"
 )
 
 func assertEqual(t *testing.T, actual, expected interface{}) {
+	t.Helper()
 	if !reflect.DeepEqual(actual, expected) {
-		t.Errorf("actual differs from expected:\n%s", diff.ObjectReflectDiff(expected, actual))
+		t.Errorf("actual differs from expected:\n%s", cmp.Diff(expected, actual, cmp.AllowUnexported(httpResult{})))
```
Comment (hongkailiu) [inline cmd/autoowners/main_test.go:20]: AllowUnexported: TIL.
I hit a similar case before and did some clumsy stuff.

File: cmd/autoowners/main.go [hunk 2]
```diff
@@ -259,7 +261,7 @@ func pullOwners(gc github.Client, configRootDir string, blocklist blocklist, con
 
 	for _, orgRepo := range orgRepos {
 		logrus.WithField("orgRepo", orgRepo.repoString()).Info("handling repo ...")
-		httpResult, err := getOwnersHTTP(gc, orgRepo)
+		httpResult, err := getOwnersHTTP(gc, orgRepo, pc.OwnersFilenames(orgRepo.Organization, orgRepo.Repository))
```
Comment (hongkailiu) [inline cmd/autoowners/main.go:264]: `pc.OwnersFilenames(orgRepo.Organization, orgRepo.Repository)` hopefully never returns us things like empty strings.

**Review by hongkailiu — COMMENTED** [review]
/lgtm

File: cmd/autoowners/main.go (hunk 2 above)
Comment (alvaroaleman) [reply cmd/autoowners/main.go:264]: No it doesn't: https://github.com/openshift/ci-tools/blob/75fa6a905caa4d0b49ce73c08fa544263c9fbab8/vendor/k8s.io/test-infra/prow/plugins/config.go#L188-L197

**Comment by alvaroaleman** [comment]
ref https://issues.redhat.com/browse/DPTP-1907

**Pull request merged**